### PR TITLE
unit tests for sealed interfaces

### PR DESCRIPTION
**Pull request by williamboxhall**

### CODE REVIEW

File: src/test/kotlin/com/cultureamp/common/ClassReflectionTest.kt [hunk 1]
```diff
@@ -166,3 +174,7 @@ sealed class FSubSubSubSubSub : FSubSubSubSub()
 data class F6(val value: String) : FSubSubSubSubSub()
 sealed class FSubSubSubSubSubSub : FSubSubSubSubSub()
 data class F7(val value: String) : FSubSubSubSubSubSub()
+
+sealed interface G
+data class G1(val value: String) : G
+data class G2(val value: Int) : G
```
Comment (davidwheeler123) [inline src/test/kotlin/com/cultureamp/common/ClassReflectionTest.kt:180]: reading up on this it seems you used to be able to only use sealed subclasses if they were in the same file. worth putting them into separate files just to be sure?

**Review by davidwheeler123 — APPROVED** [review]

**Comment by davidwheeler123** [comment]
Worth bumping a version? even if we don't publish it?

**Comment by williamboxhall** [comment]
Hmm I don't think we need a version bump.

File: src/test/kotlin/com/cultureamp/common/ClassReflectionTest.kt (hunk 1 above)
Comment (williamboxhall) [reply src/test/kotlin/com/cultureamp/common/ClassReflectionTest.kt:180]: True - let's do it as another PR though before I lose track of this :P

**Pull request merged**